### ISSUE TITLE
txnbuild: method that returns xdr.TransactionEnvelope

### ISSUE THREAD

**Issue by poliha** [stellar/go#1415]
It will be good to have  method that returns the `xdr.TransactionEnvelope` since this is not exported. There are times when you need this and it will save users from the need to marshal the base64. 

One use case is for the compliance server which sends `xdr.Transaction`. in order to get this I had to do the following
```golang
...
// error checks omitted
	tx.Build()
	tx.Sign()
	txeB64, _ := tx.Base64()

	var txXDR xdr.TransactionEnvelope
	_ = xdr.SafeUnmarshalBase64(txeB64, &txXDR)
	txB64, _ := xdr.MarshalBase64(tx)
```
This could have been
```golang
 ...
       tx.Build()
       txEnv :=  tx.TxEnvelope() // returns xdr.TransactionEnvelope
      // txEnv.Tx will contain xdr.Transaction
```

**Comment by ire-and-curses** [comment]
I like this idea a lot. It's related to #956. You could use this to export an unsigned envelope, for signing later. Would be great to coordinate the two features.

**Comment by poliha** [comment]
closed in #1438
